std/file.d

```diff
@@ -293,10 +293,17 @@ version (Posix) private void[] readImpl(const(char)[] name, const(FSChar)* namez
                 min(result.length, upTo) - size);
         cenforce(actual != -1, name, namez);
         if (actual == 0) break;
-        size += actual;
+
+        import core.checkedint : addu;
+        bool overflow;
+        size = addu(size, actual, overflow);
+        if (overflow) assert(0);
+
         if (size >= upTo) break;
         if (size < result.length) continue;
-        immutable newAlloc = size + sizeIncrement;
+        immutable newAlloc = addu(size, sizeIncrement, overflow);
+        if (overflow) assert(0);
+
         result = GC.realloc(result.ptr, newAlloc, GC.BlkAttr.NO_SCAN)[0 .. newAlloc];
```

Comment:
> You'll always crash before reaching that overflow.

Famous last words :-)

All assert's are guaranteed to never be tripped. Of course, we're often wrong about that, and they get tripped. Hence adding this here.